lib/std/heap.zig

```diff
@@ -8,6 +8,273 @@ const c = std.c;
+    .uefi => 4 << 10,
+    .freestanding => switch (builtin.cpu.arch) {
+        .wasm32, .wasm64 => 64 << 10,
+        .x86, .x86_64 => 4 << 10,
+        .thumb, .thumbeb, .arm, .armeb, .aarch64, .aarch64_be => 64 << 10,
+        // Explicitly only 4kb.
+        // https://refspecs.linuxbase.org/ELF/zSeries/lzsabi0_zSeries.html#AEN798
+        .s390x => 4 << 10,
+        // Source: Zig's own libc page.h for arc.
+        .arc => 16 << 10,
+        // Source: Wikipedia "Page (computer memory)"
+        .powerpc, .powerpc64, .powerpc64le, .powerpcle => 64 << 10,
+        .riscv32, .riscv64 => 4 << 10,
+        .sparc => 256 << 10,
+        .sparc64 => 64 << 10,
+        else => missing_max_page_size,
+    },
+    .freebsd => switch (builtin.cpu.arch) {
+        .x86, .x86_64 => 4 << 10,
+        // https://github.com/freebsd/freebsd-src/blob/0a9d1da6e6cede5e9c0ff63240d724049ad72b5b/sys/arm64/arm64/locore.S#L931
+        .thumb, .thumbeb, .arm, .armeb, .aarch64, .aarch64_be => 16 << 10,
+        .powerpc, .powerpc64, .powerpc64le, .powerpcle => missing_max_page_size,
+        .riscv32, .riscv64 => missing_max_page_size,
+        else => missing_max_page_size,
+    },
+    .netbsd => switch (builtin.cpu.arch) {
+        .x86, .x86_64 => 4 << 10,
+        // Alpha
+        .thumb, .thumbeb, .arm, .armeb, .aarch64, .aarch64_be => missing_max_page_size,
+        // https://github.com/NetBSD/src/blob/bd34fa0e7331e54a77f1d095e88c69f77ad222bd/sys/arch/mips/include/vmparam.h#L73
+        .mips, .mipsel, .mips64, .mips64el => 16 << 10,
+        // https://github.com/NetBSD/src/blob/bd34fa0e7331e54a77f1d095e88c69f77ad222bd/sys/arch/powerpc/include/vmparam.h#L41
+        .powerpc, .powerpc64, .powerpc64le, .powerpcle => 16 << 10,
+        // https://github.com/NetBSD/src/blob/bd34fa0e7331e54a77f1d095e88c69f77ad222bd/sys/arch/sparc/include/vmparam.h#L69
+        .sparc => 8 << 10,
+        .sparc64 => missing_max_page_size,
+        .riscv32, .riscv64 => missing_max_page_size,
+        // https://github.com/NetBSD/src/blob/bd34fa0e7331e54a77f1d095e88c69f77ad222bd/sys/arch/m68k/include/vmparam.h#L70
+        .m68k => 8 << 10,
+        else => missing_max_page_size,
+    },
+    .dragonfly => switch (builtin.cpu.arch) {
+        .x86, .x86_64 => 4 << 10,
+        else => missing_max_page_size,
+    },
+    .openbsd => switch (builtin.cpu.arch) {
+        .x86, .x86_64 => 4 << 10,
+        // Alpha
+        .thumb, .thumbeb, .arm, .armeb, .aarch64, .aarch64_be => missing_max_page_size,
+        .mips, .mipsel, .mips64, .mips64el, .loongarch32, .loongarch64 => missing_max_page_size,
+        .powerpc, .powerpc64, .powerpc64le, .powerpcle => missing_max_page_size,
+        .riscv64 => missing_max_page_size,
+        .sparc64 => missing_max_page_size,
+        else => missing_max_page_size,
+    },
+    .linux => switch (builtin.cpu.arch) {
+        .x86, .x86_64 => 4 << 10,
+        .thumb, .thumbeb, .arm, .armeb, .aarch64, .aarch64_be => 64 << 10,
```

Comment:
For linux/arm, the largest configurable page size I see is 4K, i.e. same as the minimum.

This is correct for linux/aarch64, however.